### PR TITLE
[api-extractor] Fix inconsistent sorting of APIs on older NodeJS versions

### PR DESCRIPTION
**Pull request by octogonz**
Fixes https://github.com/microsoft/rushstack/issues/1552

### CODE REVIEW

File: apps/rush-lib/src/logic/taskRunner/TaskCollection.ts [hunk 1]
```diff
@@ -105,9 +106,7 @@ export class TaskCollection {
     });
 
     // Sort the queue in descending order, nothing will mess with the order
-    buildQueue.sort((taskA: ITask, taskB: ITask): number => {
-      return taskB.criticalPathLength! - taskA.criticalPathLength!;
-    });
+    Sort.sortBy(buildQueue, (task: ITask): number => -task.criticalPathLength!);
```
Comment (octogonz) [inline apps/rush-lib/src/logic/taskRunner/TaskCollection.ts]: (This is probably the riskiest logic change.)

File: common/changes/@microsoft/api-extractor-model/octogonz-stable-sort_2019-11-02-20-23.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/api-extractor-model",
+      "comment": "Fix an issue where API reports sometimes were ordered differently depending",
```
Comment (iclanton) [inline common/changes/@microsoft/api-extractor-model/octogonz-stable-sort_2019-11-02-20-23.json]: Unfinished sentence?

File: libraries/node-core-library/src/LegacyAdapters.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See LICENSE in the project root for license information.
 
+import { sort as timsort } from 'timsort';
```
Comment (iclanton) [inline libraries/node-core-library/src/LegacyAdapters.ts]: It'd be cool if we could somehow only install this dependency on Node versions below 11.

File: apps/rush-lib/src/logic/taskRunner/TaskCollection.ts (hunk 1 above)
Comment (iclanton) [reply apps/rush-lib/src/logic/taskRunner/TaskCollection.ts]: The logic looks seems the same here.

File: libraries/node-core-library/src/LegacyAdapters.ts (hunk 3 above)
Comment (octogonz) [reply libraries/node-core-library/src/LegacyAdapters.ts]: I don't know of any easy way to do that. (?)

BTW `timsort` itself is tiny and has no dependencies.

File: common/changes/@microsoft/api-extractor-model/octogonz-stable-sort_2019-11-02-20-23.json (hunk 2 above)
Comment (octogonz) [reply common/changes/@microsoft/api-extractor-model/octogonz-stable-sort_2019-11-02-20-23.json]: I copy+pasted the other message into the CLI; must have gotten truncated

File: libraries/node-core-library/src/LegacyAdapters.ts (hunk 3 above)
Comment (iclanton) [reply libraries/node-core-library/src/LegacyAdapters.ts]: Probably not worth worrying about then.

**Comment by octogonz** [comment]
@mikeharder  This fix was published with API Extractor 7.5.4

**Pull request merged**